Mathlib/Order/LiminfLimsup.lean

```diff
@@ -294,6 +294,66 @@ theorem IsCobounded.mono (h : f ≤ g) : f.IsCobounded r → g.IsCobounded r
 
 end Relation
 
+section add_and_sum
+
+open Filter BigOperators Set
+
+variable {α : Type*} {f : Filter α} [NeBot f]
+variable {R : Type*} [Preorder R]
+
+lemma isBoundedUnder_ge_add [Add R]
+    [CovariantClass R R (fun a b ↦ a + b) (· ≤ ·)] [CovariantClass R R (fun a b ↦ b + a) (· ≤ ·)]
+    {u v : α → R} (u_bdd_ge : f.IsBoundedUnder (· ≥ ·) u) (v_bdd_ge : f.IsBoundedUnder (· ≥ ·) v) :
+    f.IsBoundedUnder (· ≥ ·) (u + v) := by
+  obtain ⟨U, hU⟩ := u_bdd_ge
+  obtain ⟨V, hV⟩ := v_bdd_ge
+  use U + V
+  simp only [eventually_map, Pi.add_apply] at hU hV ⊢
+  filter_upwards [hU, hV] with a hu hv using add_le_add hu hv
+
+lemma isBoundedUnder_le_add [Add R]
+    [CovariantClass R R (fun a b ↦ a + b) (· ≤ ·)] [CovariantClass R R (fun a b ↦ b + a) (· ≤ ·)]
+    {u v : α → R} (u_bdd_ge : f.IsBoundedUnder (· ≤ ·) u) (v_bdd_ge : f.IsBoundedUnder (· ≤ ·) v) :
+    f.IsBoundedUnder (· ≤ ·) (u + v) := by
+  obtain ⟨U, hU⟩ := u_bdd_ge
+  obtain ⟨V, hV⟩ := v_bdd_ge
+  use U + V
+  simp only [eventually_map, Pi.add_apply] at hU hV ⊢
+  filter_upwards [hU, hV] with a hu hv using add_le_add hu hv
+
+lemma isBoundedUnder_sum {κ : Type*} [DecidableEq κ] [AddCommMonoid R] {r : R → R → Prop}
+    (hr : ∀ (v₁ v₂ : α → R), f.IsBoundedUnder r v₁ → f.IsBoundedUnder r v₂
+      → f.IsBoundedUnder r (v₁ + v₂)) (hr₀ : r 0 0)
+    {u : κ → α → R} (s : Finset κ) :
+    (∀ k ∈ s, f.IsBoundedUnder r (u k)) →
+      f.IsBoundedUnder r (∑ k ∈ s, u k) := by
+  apply Finset.induction_on (p := fun s ↦
```

Comment:
Have you tried this with `induction`?

Reply:
Thanks!

`induction` did allow me to specify `Finset.induction_on` as the induction principle, but I could not get it to name the variables and the induction hypothesis (which is pretty annoying, since the induction hypothesis is surely needed in the proof). However, `induction'` just worked. This is definitely cleaner!